homeassistant/components/overkiz/update.py

```diff
@@ -0,0 +1,113 @@
+"""Support for Overkiz updates."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyoverkiz.enums import UIClass, UIWidget, UpdateBoxStatus
+from pyoverkiz.models import Gateway
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN
+from .entity import OverkizEntity
+
+UPDATE_COMMAND = "update"
+
+UPDATE_AVAILABLE_STATUS = {
+    UpdateBoxStatus.READY_TO_UPDATE,
+    UpdateBoxStatus.READY_TO_BE_UPDATED_BY_SERVER,
+    UpdateBoxStatus.READY_TO_UPDATE_LOCALLY,
+    UpdateBoxStatus.READY_TO_UPDATE_LOCALLY,
+}
+
+
+async def async_setup_entry(
```

Comment:
The gateway information is not refreshed in our DataUpdateCoordinator and I am not sure if the new data is broadcasted via events. We would need either to enhance our DataUpdateCoordinator, or add a second one (with different polling interval) or just add a time interval function that will poll every day.